src/System.EventLog/src/System/Diagnostics/SharedUtils.cs

```diff
@@ -0,0 +1,335 @@
+
+            EnterMutexWithoutGlobal(mutexName, ref mutex);
+        }
+
+        internal static void EnterMutexWithoutGlobal(string mutexName, ref Mutex mutex)
+        {
+            bool createdNew;
+            MutexSecurity sec = new MutexSecurity();
+            SecurityIdentifier everyoneSid = new SecurityIdentifier(WellKnownSidType.AuthenticatedUserSid, null);
+            //sec.AddAccessRule(new MutexAccessRule(everyoneSid, MutexRights.Synchronize | MutexRights.Modify, AccessControlType.Allow));
+            //Mutex tmpMutex = new Mutex(false, mutexName, out createdNew, sec);
+            Mutex tmpMutex = new Mutex(false, mutexName, out createdNew);
+            SafeWaitForMutex(tmpMutex, ref mutex);
+        }
+
+        private static bool SafeWaitForMutex(Mutex mutexIn, ref Mutex mutexOut)
+        {
+            Debug.Assert(mutexOut == null, "You must pass in a null ref Mutex");
+            while (true)
+            {
+                if (!SafeWaitForMutexOnce(mutexIn, ref mutexOut))
+                    return false;
+                if (mutexOut != null)
+                    return true;
+                Thread.Sleep(0);
+            }
+        }
+
+        [MethodImplAttribute(MethodImplOptions.NoInlining)]
+        private static bool SafeWaitForMutexOnce(Mutex mutexIn, ref Mutex mutexOut)
+        {
+            bool ret;
+            RuntimeHelpers.PrepareConstrainedRegions();
+            try
+            { }
+            finally
+            {
+                Thread.BeginCriticalRegion();
+                Thread.BeginThreadAffinity();
+                int result = WaitForSingleObjectDontCallThis(mutexIn.SafeWaitHandle, 500);
+                switch (result)
+                {
+                    case NativeMethods.WAIT_OBJECT_0:
+                    case NativeMethods.WAIT_ABANDONED:
+                        mutexOut = mutexIn;
+                        ret = true;
+                        break;
+
+                    case NativeMethods.WAIT_TIMEOUT:
+                        ret = true;
+                        break;
+
+                    default:
+                        ret = false;
+                        break;
+                }
+                if (mutexOut == null)
+                {
+                    Thread.EndThreadAffinity();
+                    Thread.EndCriticalRegion();
+                }
+            }
+
+            return ret;
+        }
+
+        [DllImport(Interop.Libraries.Kernel32, ExactSpelling = true, SetLastError = true, EntryPoint = "WaitForSingleObject")]
+        private static extern int WaitForSingleObjectDontCallThis(SafeWaitHandle handle, int timeout);
+        internal static string GetLatestBuildDllDirectory(string machineName)
+        {
+            string dllDir = "";
+            RegistryKey baseKey = null;
+            RegistryKey complusReg = null;
+            RegistryPermission registryPermission = new RegistryPermission(PermissionState.Unrestricted);
```

Comment:
remove RegistryPermission entirely from the code please, and all calls on it.

Reply:
Removed